File: c/2/1182.c
#include <stdio.h>

int main(void)
{
    int count, target_sum;
    scanf("%d %d", &count, &target_sum);

    int numbers[20];
    for (int i = 0; i < count; i++)
    {
        scanf("%d", &numbers[i]);
    }

    int result = 0;
    int sum;

    for (int i = 1; i < 1 << count; i++)
    {
        sum = 0;

        int combination = i;
        int j = 0;
        while (combination)
        {
            if (combination & 1)
                sum += numbers[j];
            combination >>= 1;
            j++;
        }

        if (sum == target_sum)
        {
            result++;
        }
    }

    printf("%d\n", result);

    return 0;
}

File: python/2/1182.py
from sys import stdin

input = stdin.readline


def combinations(length: int):
    result = list()
    for i in range(2**length):
        result.clear()
        j = 0
        while i:
            if i & 1:
                result.append(j)
            i >>= 1
            j += 1
        yield result


def main():
    count, target_sum = map(int, input().split())
    numbers = list(map(int, input().split()))

    result = 0
    for combination in combinations(count):
        if not combination:
            continue
        if sum(map(lambda x: numbers[x], combination)) == target_sum:
            result += 1

    print(result)


if __name__ == '__main__':
    main()
